<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SquareArrayDialog</class>
 <widget class="QDialog" name="SquareArrayDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>584</width>
    <height>645</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Create a Square Array</string>
  </property>
  <property name="toolTip">
   <string/>
  </property>
  <property name="toolTipDuration">
   <number>-1</number>
  </property>
  <property name="statusTip">
   <string/>
  </property>
  <property name="whatsThis">
   <string/>
  </property>
  <property name="accessibleName">
   <string/>
  </property>
  <property name="accessibleDescription">
   <string/>
  </property>
  <property name="locale">
   <locale language="English" country="UnitedKingdom"/>
  </property>
  <property name="windowFilePath">
   <string/>
  </property>
  <property name="sizeGripEnabled">
   <bool>false</bool>
  </property>
  <property name="modal">
   <bool>false</bool>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QScrollArea" name="scrollArea">
     <property name="styleSheet">
      <string notr="true"/>
     </property>
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Sunken</enum>
     </property>
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaContent">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>562</width>
        <height>557</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QTabWidget" name="tabWidget">
         <property name="currentIndex">
          <number>0</number>
         </property>
         <widget class="QWidget" name="tab">
          <attribute name="title">
           <string>Parameters</string>
          </attribute>
          <layout class="QVBoxLayout" name="verticalLayout_9">
           <item>
            <widget class="QGroupBox" name="libraryGroup">
             <property name="styleSheet">
              <string notr="true"/>
             </property>
             <property name="title">
              <string>Spectral Library</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_4">
              <item>
               <widget class="QgsFileWidget" name="browseLibrary" native="true">
                <property name="toolTip">
                 <string>Browse for the spectral library</string>
                </property>
                <property name="dialogTitle" stdset="0">
                 <string>Select a spectral library</string>
                </property>
                <property name="filter" stdset="0">
                 <string>ENVI Spectral Library (*.sli);;All Files (*)</string>
                </property>
                <property name="defaultRoot" stdset="0">
                 <string/>
                </property>
               </widget>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout">
                <item>
                 <widget class="QLabel" name="label1">
                  <property name="toolTip">
                   <string>The reflectance scale factor is automatically detected. It is the number by which the array must be divided to get reflectance values between 0 and 1. The user can change this value here.</string>
                  </property>
                  <property name="text">
                   <string>Reflectance scale factor:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QSpinBox" name="libraryScaleValue">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="minimumSize">
                   <size>
                    <width>100</width>
                    <height>0</height>
                   </size>
                  </property>
                  <property name="toolTip">
                   <string>The reflectance scale factor is automatically detected. It is the number by which the array must be divided to get reflectance values between 0 and 1. The user can change this value here.</string>
                  </property>
                  <property name="minimum">
                   <number>1</number>
                  </property>
                  <property name="maximum">
                   <number>1000000000</number>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="horizontalSpacer">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="constraintsGroup">
             <property name="toolTip">
              <string>Only models that meet these constraints can be chosen.</string>
             </property>
             <property name="title">
              <string>Set constraints</string>
             </property>
             <property name="flat">
              <bool>false</bool>
             </property>
             <property name="checkable">
              <bool>true</bool>
             </property>
             <layout class="QGridLayout" name="gridLayout_4">
              <property name="leftMargin">
               <number>25</number>
              </property>
              <property name="topMargin">
               <number>20</number>
              </property>
              <item row="0" column="0">
               <widget class="QGroupBox" name="minFracCheck">
                <property name="styleSheet">
                 <string notr="true">QGroupBox { border: none}</string>
                </property>
                <property name="title">
                 <string>Min endmember fraction:</string>
                </property>
                <property name="alignment">
                 <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
                </property>
                <property name="flat">
                 <bool>false</bool>
                </property>
                <property name="checkable">
                 <bool>true</bool>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_13">
                 <property name="spacing">
                  <number>6</number>
                 </property>
                 <property name="leftMargin">
                  <number>25</number>
                 </property>
                 <property name="topMargin">
                  <number>20</number>
                 </property>
                 <item>
                  <widget class="QDoubleSpinBox" name="minFracValue">
                   <property name="readOnly">
                    <bool>false</bool>
                   </property>
                   <property name="buttonSymbols">
                    <enum>QAbstractSpinBox::UpDownArrows</enum>
                   </property>
                   <property name="minimum">
                    <double>-0.500000000000000</double>
                   </property>
                   <property name="maximum">
                    <double>1.000000000000000</double>
                   </property>
                   <property name="singleStep">
                    <double>0.010000000000000</double>
                   </property>
                   <property name="value">
                    <double>-0.050000000000000</double>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item row="0" column="2">
               <widget class="QGroupBox" name="maxRMSECheck">
                <property name="styleSheet">
                 <string notr="true">QGroupBox { border: none  }</string>
                </property>
                <property name="title">
                 <string>Max RMSE:</string>
                </property>
                <property name="checkable">
                 <bool>true</bool>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_12">
                 <property name="leftMargin">
                  <number>25</number>
                 </property>
                 <property name="topMargin">
                  <number>20</number>
                 </property>
                 <item>
                  <widget class="QDoubleSpinBox" name="maxRMSEValue">
                   <property name="decimals">
                    <number>3</number>
                   </property>
                   <property name="maximum">
                    <double>0.100000000000000</double>
                   </property>
                   <property name="singleStep">
                    <double>0.001000000000000</double>
                   </property>
                   <property name="value">
                    <double>0.025000000000000</double>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QGroupBox" name="maxFracCheck">
                <property name="styleSheet">
                 <string notr="true">QGroupBox { border: none}</string>
                </property>
                <property name="title">
                 <string>Max endmember fraction:</string>
                </property>
                <property name="checkable">
                 <bool>true</bool>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_5">
                 <property name="leftMargin">
                  <number>25</number>
                 </property>
                 <property name="topMargin">
                  <number>20</number>
                 </property>
                 <item>
                  <widget class="QDoubleSpinBox" name="maxFracValue">
                   <property name="minimum">
                    <double>0.000000000000000</double>
                   </property>
                   <property name="maximum">
                    <double>1.500000000000000</double>
                   </property>
                   <property name="singleStep">
                    <double>0.010000000000000</double>
                   </property>
                   <property name="value">
                    <double>1.050000000000000</double>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item row="1" column="0" colspan="3">
               <widget class="QGroupBox" name="resetCheck">
                <property name="styleSheet">
                 <string notr="true">QGroupBox { border: none}</string>
                </property>
                <property name="title">
                 <string>Reset fractions to threshold</string>
                </property>
                <property name="checkable">
                 <bool>true</bool>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_6">
                 <property name="leftMargin">
                  <number>25</number>
                 </property>
                 <property name="topMargin">
                  <number>20</number>
                 </property>
                 <item>
                  <widget class="QLabel" name="label">
                   <property name="text">
                    <string>Fractions will be reset to the threshold values when models do not meet the fraction constraints. RMSE is then calculated with these corrected values.</string>
                   </property>
                   <property name="wordWrap">
                    <bool>true</bool>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="outputGroup">
             <property name="toolTip">
              <string>Choose a name and location for the MESMA result file. Additional files will be filename_fractions, filename_rmse, filename_metadata.txt and filename_look_up_table.csv.</string>
             </property>
             <property name="title">
              <string>Output Square Array</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_3">
              <item>
               <widget class="QgsFileWidget" name="browseOut" native="true">
                <property name="toolTip">
                 <string>Choose a name and location for the MESMA result file. Additional files will be filename_fractions, filename_rmse, filename_metadata.txt and filename_look_up_table.csv.</string>
                </property>
                <property name="filter" stdset="0">
                 <string>Square Array (*.sqr);;All Files (*)</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="label_2">
                <property name="text">
                 <string>Output metrics (reduce file size):</string>
                </property>
               </widget>
              </item>
              <item>
               <layout class="QVBoxLayout" name="verticalLayout_7">
                <property name="leftMargin">
                 <number>20</number>
                </property>
                <item>
                 <widget class="QCheckBox" name="outRmseCheck">
                  <property name="text">
                   <string>RMSE (required for IES and EMC)</string>
                  </property>
                  <property name="checked">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QCheckBox" name="outConstrCheck">
                  <property name="text">
                   <string>Constraints breached (required for IES and EMC)</string>
                  </property>
                  <property name="checked">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QCheckBox" name="outAngleCheck">
                  <property name="text">
                   <string>Spectral Angle (required for EMC)</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QCheckBox" name="outFracCheck">
                  <property name="text">
                   <string>Endmember Fractions</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QCheckBox" name="outShadeCheck">
                  <property name="text">
                   <string>Shade Fraction</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <widget class="QCheckBox" name="openInQGIS">
                <property name="toolTip">
                 <string>Check if you would like to view the result in QGIS afterwards.</string>
                </property>
                <property name="text">
                 <string>Open result in QGIS</string>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="verticalSpacer">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>0</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="tab_log">
          <attribute name="title">
           <string>Log</string>
          </attribute>
          <layout class="QVBoxLayout" name="verticalLayout_10">
           <item>
            <widget class="QTextBrowser" name="logBrowser"/>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QProgressBar" name="progressBar">
     <property name="value">
      <number>0</number>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="OKClose">
     <property name="standardButtons">
      <set>QDialogButtonBox::Close|QDialogButtonBox::Ok</set>
     </property>
     <property name="centerButtons">
      <bool>false</bool>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QgsFileWidget</class>
   <extends>QWidget</extends>
   <header>qgsfilewidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
